src/SearchField/README.md

```diff
@@ -0,0 +1,88 @@
+# SearchField
+
+Provides a basic search field component.
+
+## API
+
+### `screenReaderText` (string; required)
+`screenReaderText` specifies the text for screen readers within the `label` element of the `InputText` component.
+
+### `onSubmit` (function; required)
+`onSubmit` specifies a callback function for when the `SearchField` is submitted by the user. For example:
+
+```jsx
+<SearchField
+  screenReaderText="search"
+  onSubmit={(value) => { console.log(value); }}
+/>
+```
+
+### `onChange` (function; optional)
+`onChange` specifies a callback function for when the value in `SearchField` is changed by the user. The default is an empty function. For example:
+
+```jsx
+<SearchField
+  screenReaderText="search"
+  onChange={(value) => { console.log(value); }}
+/>
+```
+
+### `onFocus` (function; optional)
+`onFocus` specifies a callback function for when the value in `SearchField` is changed by the user. The default is an empty function. For example:
+
+```jsx
+<SearchField
+  screenReaderText="search"
+  onFocus={(value) => { console.log(value); }}
+/>
+```
+
+### `onBlur` (function; optional)
+`onBlur` specifies a callback function for when the value in `SearchField` is changed by the user. The default is an empty function. For example:
+
+```jsx
+<SearchField
+  screenReaderText="search"
+  onBlur={(value) => { console.log(value); }}
+/>
+```
+
+### `placeholder` (string; optional)
+`placeholder` specifies the placeholder text for the input. The default is an empty string.
+
+### `value` (string; optional)
+`value` specifies the initial value for the input. The default is an empty string.
+
+### `hasClearButton` (boolean; optional)
+`hasClearButton` specifies whether the search field has an additional button to allow the user to clear the `input` value. The default is `true`.
+
+### `hasRoundedCorners` (boolean; optional)
+`hasRoundedCorners` specifies whether the search field has rounded corners. The default is `false`.
+
+### `hideInputLabel` (boolean; optional)
+`hideInputLabel` specifies whether the label for the `input` element is hidden. The default is `false`.
+
+### `inputLabel` (string; optional)
+`inputLabel` specifies the label text to use for the `input` element. The default is an empty string, which results in a search icon to be used as the `label` element.
+
+### `searchButtonLabel` (string or element; optional)
+`searchButtonLabel` specifies the text or element for the search button. For example:
+
+```jsx
+<SearchField
+    screenReaderText="search"
+    onSubmit={action('search-submitted')}
```

Comment:
Nit: Can you make these callback names more specific, like `handleOnSubmit`? If an end user is unfamiliar with `action`s they might not know what `action` means in this scenario.

Reply:
I'm not sure I follow your question here. The `action` is just for the storybook actions that get logged. I don't think changing the callback names like `handleOnSubmit` would add much clarity to an end user compared to `onSubmit`.

Reply:
Right - because `action` is specific to storybook actions, somebody reading the documentation might be like "what is this `action` method"? if they're unfamiliar with Storybook.

Whereas I think that the general name for "some generic callback method" (or at least, from the documentation I've read) is either `callback` or `someCallback` or `handleOn{someEventName}`.